Events.md

```diff
@@ -0,0 +1,225 @@
+
+Approvals are based on the pair of outbound and inbound token. Be aware that it is not offerlist bases, as an offerlist also holds the tickscale.
+
+We emit `outbound` token, `inbound` token, `owner`, msg.sender (`spender`), `value`. Where `owner` is the one who owns the funds, `spender` is the one who is allowed to use the funds and `value` is the amount of funds that is allowed to be used.
+
+Outbound, inbound and owner is indexed, this way one can filter on these fields when doing RPC calls. If we could somehow combine outbound and inbound into one field, then we could also index the spender.
+
+### Credit
+
+This is emitted when a user's account on Mangrove is credited with some native funds, to be used as provision for offers.
+
+It emits the `maker`'s address and the `value` credited
+
+The `maker` address is indexed so that we can filter on it when doing RPC calls.
+
+These are the scenarios where it can happen:
+
+- Fund Mangrove directly
+- Fund Mangrove when posting an offer
+- When updating an offer
+  - Funding Mangrove or
+  - the updated offer needs less provision, than it already has. Meaning the user gets credited the difference.
+- When retracting an offer and deprovisioning it.Meaning the user gets credited the provision that was locked by the offer.
+- When an offer fails. The remaining provision gets credited back to the maker
+
+A challenge for an indexer is to know how much provision each offer locks. With the current events, an indexer is going to have to know the liveness, gasreq and gasprice of the offer. If the offer is not live, then also know if it has been deprovisioned. And know what the gasbase of the offerlist was when the offer was posted. With this information an indexer can calculate the exact provision locked by the offer.
+
+Another information that an indexer cannot deduce, is in what scenario the credit event happened. E.g. We don't know if the credit event happened because an offer failed or if the user simply funded Mangrove.
+
+### Debit
+
+This is emitted when a user's account on Mangrove is debited with some native funds.
+
+It emits the `maker`'s address and the `value` debited.
+
+The `maker` address is indexed so that we can filter on it when doing RPC calls.
+
+These are the scenarios where it can happen:
+
+- Withdraw funds from Mangrove directly
+- When posting an offer. The user gets debited the provision that the offer locks.
+- When updating an offer and it requires more provision that it already has. Meaning the user gets debited the difference.
+
+Same challenges as for Credit
+
+### Kill
+
+Emitted when the Mangrove instance is killed
+
+### NewMgv
+
+Emitted when a new Mangrove is deployed.
+
+### OrderStart
+
+This event is emitted when a market order is started on Mangrove.
+
+It emits the `offerlist key`, the `taker`, the `maxLogPrice`, `fillVolume` and `fillWants`.
```

Comment:
offerlist => offerList in general - and it is the hash that gets emitted, right?

Reply:
Fixed